external-crates/move/crates/move-binary-format/src/compatibility.rs

```diff
@@ -269,35 +455,10 @@ impl Compatibility {
         let old_friend_module_ids: BTreeSet<_> = old_module.friends.iter().cloned().collect();
         let new_friend_module_ids: BTreeSet<_> = new_module.friends.iter().cloned().collect();
         if !old_friend_module_ids.is_subset(&new_friend_module_ids) {
-            friend_linking = false;
+            context.function_friend_missing();
```

Comment:
Note that this is not about functions missing, this is about the new module "unfriending" some other module. (This is actually not a problem on Sui though, right @tzakian ?)

```suggestion
            context.friend_module_missing(
                &old_friend_module_ids, 
                &new_friend_module_ids,
            );
```

Reply:
We talked about the friend part, the way I understood it was friend links will fail at compile time when it matters. But if a friend link doesn't exist at upgrade that is okay since it's not a public interface. Friend links can be broken since the compiler will check if its usage is also removed within the package.

Reply:
Your understanding is correct for how things work on Sui, this is related to the question I left for @tzakian in my review:

> The fact that we parametrise friend linking is adding some complexity, and IIRC we don't use it today, because friendship is always local to a package. Is it worth getting rid of that support and hardcoding friendship compatibility (As a follow-up PR)?

`Compatibility` predates our use of Move in Sui, and it also caters to integrations of Move where a package can be built up module-by-module under an address, and where those modules can be updated in place. In that setting, you care about friend linkage, because you don't want to introduce module M with friend N, introduce N which relies on M, then upgrade M to break its friendship with N.

None of this is possible on Sui though, and I would guess that we have friend linking checks disabled, so we should probably get rid of this support and make our lives easier unless @tzakian strongly objects or that might impact some future plans.